File: frontend/lib/models/user_model.dart
class UserModel {
  final String id;
  final String username;
  final String name;
  final String phone;
  final String email;
  final String? avatar;
  final int loves;

  UserModel({
    required this.id,
    required this.username,
    required this.name,
    required this.phone,
    required this.email,
    this.avatar,
    required this.loves,
  });

  factory UserModel.fromJson(Map<String, dynamic> json)
  => UserModel(
    id: json["_id"],
    username: json["username"],
    name: json["name"],
    phone: json["phone"],
    email: json["email"],
    avatar: json["avatar"] ?? '',
    loves: json["loves"],
  );
}

File: backend/Models/userModel.ts
import mongoose from 'mongoose';

const userSchema = new mongoose.Schema({
  username: {
    type: String,
    required: true,
    unique: true,
    trim: true,
    minLength: [4, 'Username must be at least 4 characters long']
  },
  name: {
    type: String,
    required: true,
    trim: true,
  },
  phone: {
    type: String,
    required: [true, 'Phone number is required'],
    unique: true,
    trim: true,
    validate: {
      validator: (value: string) => {
        return /^(\+91)?[6789]\d{9}$/.test(value);
      },
      message: props => `${props.value} is not a valid phone number!`
    }
  },
  email: {
    type: String,
    required: [true, 'Email is required'],
    trim: true,
    validate: {
      validator: (value: string) => {
        return /^([\w-\.]+@([\w-]+\.)+[\w-]{2,4})?$/.test(value);
      },
      message: props => `${props.value} is not valid!`
    }
  },
  avatar: {
    type: String,
    default: '',
  }
});

const User = mongoose.model('User', userSchema);
export default User;
